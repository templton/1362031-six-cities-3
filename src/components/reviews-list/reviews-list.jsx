import React, {PureComponent} from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";
import Review from "../review/review";
import ReviewForm from "../review-form/review-form";
import {selectAuthStatus} from "../../store/user/selectors";
import {selectReviews} from "../../store/reviews/selectors";
import {Operation as ReviewsOperation} from "../../store/reviews/reducers";

class ReviewList extends PureComponent {

  componentDidMount() {
    const {getReviews, hotelId} = this.props;
    getReviews(hotelId);
  }

  render() {
    const {reviews, authStatus, hotelId} = this.props;
    return (
      <section className="property__reviews reviews">
        <h2 className="reviews__title">Reviews &middot; <span className="reviews__amount">{reviews.length}</span></h2>
        <ul className="reviews__list">
          {reviews.map((item) => {
            return (
              <li key={item.id} className="reviews__item">
                <Review reviewDetail={item} />
              </li>
            );
          })}
        </ul>
        {
          // authStatus && <ReviewForm currentHotelId={hotelId}/>
        }
        <ReviewForm currentHotelId={hotelId}/>
      </section>
    );
  }
}

ReviewList.propTypes = {
  reviews: PropTypes.array.isRequired,
  authStatus: PropTypes.bool.isRequired,
  hotelId: PropTypes.number.isRequired,
  getReviews: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => ({
  authStatus: selectAuthStatus(state),
  reviews: selectReviews(state)
});

const mapDispatchToProps = ({
  getReviews: ReviewsOperation.getReviews
});

export {ReviewList};
export default connect(mapStateToProps, mapDispatchToProps)(ReviewList);
