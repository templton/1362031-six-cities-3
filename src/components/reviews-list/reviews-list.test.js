import React from "react";
import renderer from "react-test-renderer";
import {ReviewList} from "./reviews-list";

describe(`Render <ReviewList/>`, () => {
  it(`Render no auth`, () => {
    const tree = renderer
      .create(<ReviewList
        reviews={[]}
        authStatus={false}
        hotelId={1}
        getReviews={()=>([])}/>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
