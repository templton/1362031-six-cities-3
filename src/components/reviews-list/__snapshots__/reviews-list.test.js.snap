// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render <ReviewList/> empty render 1`] = `
<section
  className="property__reviews reviews"
>
  <h2
    className="reviews__title"
  >
    Reviews Â· 
    <span
      className="reviews__amount"
    >
      0
    </span>
  </h2>
  <ul
    className="reviews__list"
  />
</section>
`;
